src/class-wpml-elementor-translatable-nodes.php

```diff
@@ -106,10 +86,10 @@ public function update( $node_id, $element, WPML_PB_String $string ) {
 					$field_key = $field['field'];
 
 					if ( $this->get_string_name( $node_id, $field, $element ) === $string->get_name() ) {
-						if ( is_numeric( $key ) ) {
-							$element[ $this->settings_field ][ $field_key ] = $string->get_value();
+						if ( isset( $element[ self::SETTINGS_FIELD ][ $field_key ] ) && is_string( $element[ self::SETTINGS_FIELD ][ $field_key ] ) ) {
```

Comment:
@strategio this looks almost the same as like #37, but is missing the `trim()`: is that in purpose?

If not, that would make my suggestion to extract the code even more important, because it could be reused.

Even if purposely omitted  `trim()`, you can still extract the code in two functions:
- One (let's call it "A") which returns the result of ` isset( $element[ self::SETTINGS_FIELD ][ $field_key ] ) && is_string( $element[ self::SETTINGS_FIELD ][ $field_key ] )`
- Another one (let's call it "B") which returns `A( $field_key ) && trim( $element[ self::SETTINGS_FIELD ][ $field_key ] )`

Reply:
@andreasciamanna, I think the `trim` check is not really needed when we update because we already had this checked when we created the string (if the string is empty, it won't be registered). However, I think it's better to have a symetric logic here to make it more readable and consistent. So I reused the same conditions (`is_flat_field` and `is_array_field`) and tests are still passing.